### DISCUSSION TITLE
Custom seeds give "Found degenerate configuration"

### DISCUSSION THREAD

**Discussion by whishei** [neperfepx/neper#188] [Q&A] [answered]
Hi,

I am trying to generate a radially-orientated particle, but I am running into an issue providing the seeds. 

This tessellation works fine:
$ neper -T -n 4 -dim 2 -id 1 -domain "circle(1)" -morphooptiini "coo:file(coo2.txt)" -o Test4
where coo2.txt contains:
0.25 0.25
0.15 0.75
0.75 0.25
0.75 0.75
![image](https://user-images.githubusercontent.com/51369978/114909982-926d4c80-9dda-11eb-8837-df539c70466d.png)

But this tessellation produces the following error:
$ neper -T -n 4 -dim 2 -id 1 -domain "circle(1)" -morphooptiini "coo:file(coo2.txt)" -o Testing
where coo2.txt is the file attached - 
[coo2.txt](https://github.com/rquey/neper/files/6319967/coo2.txt)

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 18 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/swhitman/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 4 -dim 2 -id 1 -domain circle(1) -morphooptiini
         coo:file(coo2.txt) -o Testing
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Error  :     > Found degenerate configuration.
/var/spool/slurmd/job6646196/slurm_script: line 49: 58224 Aborted                 (core dumped) neper -T -n 4 -dim 2 -id 1 -domain "circle(1)" -morphooptiini "coo:file(coo2.txt)" -o Testing

**Comment by rquey** [comment, marked as answer]
That is a degenerate configuration where (in 2D) more than 3 cells meet at a given point. To solve the problem, slightly randomize the coordinates.

**Reply by Mikewangkang** [reply]
Hi， @rquey 
Which command line is used to achieve slight coordinate randomness?

**Reply by rquey** [reply]
There is no Neper command for this. So, do it externally...
